contracts/ProjectRegistry.sol

```diff
@@ -0,0 +1,262 @@
+// Copyright (C) 2020-2022 SubProject Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity 0.8.15;
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+
+import './interfaces/IIndexerRegistry.sol';
+import './interfaces/IStaking.sol';
+import './interfaces/ISettings.sol';
+import './interfaces/IProjectRegistry.sol';
+import './interfaces/IServiceAgreementRegistry.sol';
+
+/**
+ * @title Project Registry Contract
+ * @notice ### Overview
+ * This contract tracks all projects and their deployments. At the beginning of the network,
+ * we will start with the restrict mode which only allow permissioned account to create and update project.
+ * Indexers are able to start and stop indexing with a specific deployment from this conttact. Also Indexers can update and report
+ * their indexing status from this contarct.
+ */
+contract ProjectRegistry is Initializable, OwnableUpgradeable, ERC721Upgradeable, IProjectRegistry {
+    /// @notice project information
+    struct ProjectInfo {
+        uint256 projectId;
+        address owner;
+        bytes32 latestVersion;
+        bytes32 latestDeploymentId;
+        bytes32 metadata;
```

Comment:
can we use `erc-721's metadata extension` for metadata?

Reply:
We can use metadata from ERC-721

Reply:
```
struct ProjectInfo {
  bytes32 latestDeploymentId;
  ProjectType projectType;
}
struct DeploymentInfo {
  uint256 projectId
  bytes32 metadata // aka version
}
```